### PR TITLE
[WIP] feat: click trigger treemap tooltip

### PR DESCRIPTION
**Pull request by ckifer**
## Description

Realized that Treemap currently has no on-click Tooltip ability. Hacked this together quickly, adding as draft for now. Watch video for behavior

TODO: TESTS!!

## Related Issue

https://github.com/recharts/recharts/issues/4238

## Motivation and Context

Feature parity with other charts

## How Has This Been Tested?

- manually so far
- TODO: unit tests

## Screenshots (if appropriate):

https://github.com/recharts/recharts/assets/25180830/1475b673-1a14-4ab3-a339-e33496d4755c

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] I have added a storybook story or extended an existing story to show my changes
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by PavelVanecek** [comment]
Hahaaa I like the bot harassing on new lines without coverage

File: src/chart/Treemap.tsx [hunk 1]
```diff
@@ -360,8 +360,9 @@ export class Treemap extends PureComponent<Props, State> {
     e.persist();
     const { onMouseEnter, children } = this.props;
     const tooltipItem = findChildByType(children, Tooltip);
+    const tooltipTrigger = tooltipItem.props.trigger;
```
Comment (PavelVanecek) [inline src/chart/Treemap.tsx:363]: Another `findChildByType` 😢 we need to move all of this to context

Comment (PavelVanecek) [reply src/chart/Treemap.tsx:363]: Outside of this PR of course.

**Comment by ckifer** [comment]
LOL, bad me for not having enough time for tests at midnight my time 🤣

File: src/chart/Treemap.tsx [hunk 2]
```diff
@@ -419,8 +421,30 @@ export class Treemap extends PureComponent<Props, State> {
...
   handleClick(node: TreemapNode) {
-    const { onClick, type } = this.props;
-    if (type === 'nest' && node.children) {
+    const { onClick, type, children } = this.props;
+    const tooltipItem = findChildByType(children, Tooltip);
+    const tooltipTrigger = tooltipItem.props.trigger;
+    const { activeNode } = this.state;
+    const isNestedParentNode = type === 'nest' && node.children;
+
+    if (tooltipTrigger === 'click') {
```
Comment (PavelVanecek) [inline src/chart/Treemap.tsx:430]: How much of this is copy-paste from `generateCategoricalChart`? Just wondering.

**Review by PavelVanecek — COMMENTED** [review]
I wonder if I can convince you to move the tooltip behaviour to context while working on the Treemap onClick fix.

File: src/chart/Treemap.tsx (hunk 2 above)
Comment (ckifer) [reply src/chart/Treemap.tsx:430]: I just hacked this to get it it to work, no references from anywhere except Treemap

File: src/chart/Treemap.tsx (hunk 1 above)
Comment (ckifer) [reply src/chart/Treemap.tsx:363]: yep. I'm solving problems from issues in a siloed way.. need to start helping with the broader changes.

**Comment by ckifer** [comment]
I'll take a step back from this change and try to narrow in on moving the Tooltip to context instead. This functionality was worth a mini-POC either way. Definitely something to come back to

**Pull request closed without merging**